test/src/test/java/org/corfudb/integration/MergeTestListenerIT.java

```diff
@@ -0,0 +1,402 @@
+        openMergedTable(sinkCorfuStores.get(SINK2_INDEX));
+
+        writeDataOnSource(0, NUM_WRITES);
+
+        startReplicationServers();
+
+        verifySessionInLogEntrySyncState(SINK2_INDEX, SessionManager.getDefaultLogicalGroupSubscriber());
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), NUM_WRITES, sinkTablesGroupA);
+
+        writeDataOnSink(NUM_WRITES, NUM_WRITES);
+
+        MergeTestListener listenerSink2 = new MergeTestListener(sinkCorfuStores.get(SINK2_INDEX)
+                , NAMESPACE, numTables,
+                numTables);
+
+        // Subscribe the listeners on the sink side
+        sinkCorfuStores.get(SINK2_INDEX).subscribeLogReplicationListener(listenerSink2,
+                NAMESPACE, STREAMING_TAG, sinkCorfuStores.get(SINK2_INDEX));
+
+        // Verify the listeners get all the update
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 2 * NUM_WRITES,
+                        Collections.singletonList(mergedTable));
+
+        writeDataOnSource(2*NUM_WRITES, NUM_WRITES);
+
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), 3 * NUM_WRITES, sinkTablesGroupA);
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 3 * NUM_WRITES,
+                Collections.singletonList(mergedTable));
+
+        writeDataOnSink(3*NUM_WRITES, NUM_WRITES);
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES,
+                Collections.singletonList(mergedTable));
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES, sinkTablesGroupA);
+
+        logicalGroupClient.addDestinations(GROUP_A,
+                Collections.singletonList(DefaultClusterConfig.getSinkClusterIds().get(SINK3_INDEX)));
+        List<Table<Sample.StringKey, SampleSchema.SampleGroupMsgA, Message>> sinkTablesGroupAOnSink3 = new ArrayList<>();
+        openGroupATable(numTables, sinkCorfuStores.get(SINK3_INDEX), sinkTablesGroupAOnSink3);
+
+        writeDataOnSource(NUM_WRITES, NUM_WRITES);
+
+        verifyGroupATableData(sinkCorfuStores.get(SINK3_INDEX), 3 * NUM_WRITES, sinkTablesGroupAOnSink3);
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES, sinkTablesGroupA);
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES,
+                Collections.singletonList(mergedTable));
+
+        logicalGroupClient.removeDestinations(GROUP_A,
+                Collections.singletonList(DefaultClusterConfig.getSinkClusterIds().get(SINK3_INDEX)));
+
+        writeDataOnSource(3*NUM_WRITES, NUM_WRITES);
+
+        verifyGroupATableData(sinkCorfuStores.get(SINK3_INDEX), 3 * NUM_WRITES, sinkTablesGroupAOnSink3);
+        verifyGroupATableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES, sinkTablesGroupA);
+        verifyMergedTableData(sinkCorfuStores.get(SINK2_INDEX), 4 * NUM_WRITES,
+                Collections.singletonList(mergedTable));
```

Comment:
it's a good start and covers most happy path, thanks! Please add comments for important steps. Also, to cover the whole happy path we need to consider DELETE case (where computing diff came in for `processUpdatesInSnapshotSync()`):
* For local table DELETE, it just passes through and applied to MERGE table
* For LR table DELETE on Source, we compute the diff and apply in `onSnapshotSyncEnd()`